lib/modules/datasource/common.ts

```diff
@@ -181,14 +181,36 @@ export function applyConstraintsFiltering<
         continue;
       }
 
-      const satisfiesConstraints = constraint.some(
-        // If the constraint value is a subset of any release's constraints, then it's OK
-        // fallback to release's constraint match if subset is not supported by versioning
-        (releaseConstraint) =>
-          !releaseConstraint ||
-          (versioning.subset?.(configConstraint, releaseConstraint) ??
-            versioning.matches(configConstraint, releaseConstraint)),
-      );
+      let satisfiesConstraints = false;
+      for (const releaseConstraint of constraint) {
+        if (!releaseConstraint) {
+          satisfiesConstraints = true;
+          break;
+        }
+
+        // istanbul ignore if: defensive programming
+        if (!versioning.isValid(releaseConstraint)) {
+          logger.once.debug(
+            {
+              packageName: config.packageName,
+              versioning: versioningName,
+              constraint: releaseConstraint,
+            },
+            'Invalid release constraint',
+          );
+          continue;
+        }
```

Comment:
I'm not even sure it's the proposed behavior is optimal. It could reject version, or it may ignore the particular invalid constraint. Both are not optimal, but damage may vary.

Reply:
i think it's better to ignore the contraint 🤔